public/components/main/report_definition_details/report_definition_details.tsx

```diff
@@ -30,62 +30,60 @@ import {
-  report_footer: '--',
-  trigger_type: 'Schedule',
-  schedule_details: '--',
-  alert_details: '--',
-  status: 'Active',
-  delivery_channels: ['Kibana reports'],
-  kibana_recipients: ['admin'],
-  email_recipients: 'user1@email.com',
-  email_subject: 'Latest web traffic report',
-  email_body:
-    'View report details %REPORT_DETAILS_URL% Download report file %FILE_DOWNLOAD_URL%',
-  include_report_as_attachment: true,
-};
+  const handleReportDefinitionDetails = (e) => {
+    setReportDefinitionDetails(e);
+  };
 
-interface ReportDefinitionDetailsRouteProps {
-  reportDefinitionId: string;
-  reportDefinitionDetailsMetadata: {
-    name: string;
-    description: string;
-    created: string;
-    last_updated: string;
-    source: string;
-    time_period: string;
-    file_format: string;
-    report_header: string;
-    report_footer: string;
-    trigger_type: string;
-    schedule_details: string;
-    alert_details: string;
-    status: string;
-    delivery_channels: string[];
-    kibana_recipients: string[];
-    email_recipients: string;
-    email_subject: string;
-    email_body: string;
-    include_report_as_attachment: boolean;
+  const getReportDefinitionDetailsMetadata = (data) => {
+    let reportDefinitionDetails = {
+      name: data['report_name'],
+      description: data['description'],
+      created: data['time_created'],
+      lastUpdated: data['last_updated'],
+      source: data['report_source'],
+      timePeriod:
+        data['trigger']['trigger_params']['schedule']['interval']['period'] +
+        ' ' +
+        data['trigger']['trigger_params']['schedule']['interval']['unit'],
+      fileFormat: data['report_params']['report_format'],
+      reportHeader: '--',
+      reportFooter: '--',
+      triggerType: data['trigger']['trigger_type'],
+      scheduleDetails: data['trigger']['trigger_params']['schedule_type'],
+      alertDetails: '--',
+      status: data['status'],
+      deliveryChannels: data['delivery']['channel'],
+      kibanaRecipients: data['delivery']['delivery_params']['recipients'],
+      emailRecipients: '--', // todo: data model needs separate field for email vs kibana recipients
+      emailSubject: data['delivery']['delivery_params']['subject'],
+      emailBody: data['delivery']['delivery_params']['body'],
+      includeReportAsAttachment:
+        data['delivery']['delivery_params']['has_attachment'],
+    };
+    return reportDefinitionDetails;
   };
-}
 
-export function ReportDefinitionDetails(
-  props: ReportDefinitionDetailsRouteProps
-) {
-  const reportId = props.reportDefinitionId;
+  useEffect(() => {
+    const { httpClient } = props;
+    httpClient
+      .get(`../api/reporting/reportDefinitions/${reportDefinitionId}`)
+      .then((response) => {
+        handleReportDefinitionDetails(
+          getReportDefinitionDetailsMetadata(response)
+        );
+      })
+      .catch((error) => {
+        console.log('error when getting report definition details:', error);
```

Comment:
Would it be better to use console.error here?

Reply:
Sure, will change